build.sh

```diff
@@ -0,0 +1,175 @@
+#! /bin/bash
+: '
+For building Calamares using this program, in the application "Terminal" type:
+"/pathToThisFolder/build.sh" "[arguments]"
+
+Where "[arguments]" can either be:
+
+- The build type: "Release" or "Debug"
+- The building flags, for example: -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=/usr
+'
+
+here="$(realpath "$(dirname "${0}")")"
+build="${here}/_BUILD"
+
+
+mainFunction () {
+	enterBuildDir
+	configureBuild
+	compile
+}
+
+
+checkDependencies () {
+	local name=""
+	local location=""
+	local dependency=""
+	local missing=()
+
+	local dependencies=(
+		ack@/usr/bin/vendor_perl/ack
+		autoconf@/usr/bin/autoconf
+		automake@/usr/bin/automake
+		binutils@/usr/bin/addr2line
+		bison@/usr/bin/bison
+		boost-dev@/usr/include/boost/accumulators/accumulators.hpp
+		boost-python3-dev@/usr/include/boost/mpi/python.hpp
+		cmake@/usr/bin/cmake
+		extra-cmake-modules@/usr/share/ECM/cmake/ECMConfig.cmake
+		fakeroot@/usr/bin/fakeroot
+		flex@/usr/bin/flex
+		gcc@/usr/bin/gcc
```

Comment:
**A** C++ compiler is needed, but not necessarily gcc.

Reply:
[It doesn't make a difference](https://signalvnoise.com/archives2/it_just_doesnt_matter.php).